content/src/content/docs/learn/tutorials/hooking-up-a-database.mdx

```diff
@@ -0,0 +1,466 @@
+---
+type: Tutorial
+title: Hooking up a database
+tags:
+  - some
+  - tags
+sidebar:
+  order: 3
+---
+
+import { Aside } from "@astrojs/starlight/components"
+
+The next endpoint we're going to write is one that lists all of the users in a
+database. For this tutorial I'm going to use Postgres running in Docker as our
+database. If you don't have Docker installed, I recommend following the
+[official getting started guide](https://www.docker.com/get-started/).
+
+When you're ready, you can start a Postgres container with:
+
+```shell showLineNumbers=false
+docker run -d -e POSTGRES_HOST_AUTH_METHOD=trust -p 5432:5432 postgres:17
+```
+
+<Aside type="note" title="I can't use Docker!">
+If you aren't able to run Docker locally, for example because your work's laptop
+policy doesn't allow it, an alternative is to use a hosted Postgres service.
+For example, [Neon](https://neondb.com/) offers a free tier that should be
+more than enough for this tutorial. They also offer a web-based UI for creating
+tables and rows.
+</Aside>
+
+## Creating test data
+
+To create our `users` table and data, we're going to use the `psql`
+tool that's present inside of the Docker container we created. Open a new
+terminal and run:
+
+```shell showLineNumbers=false
+docker exec -it $(docker ps -qf ancestor=postgres) psql -U postgres
+```
+
+This will drop you into a `psql` shell, with a prompt that starts `postgres=#`.
+From here you can run the following SQL commands:
+
+```sql showLineNumbers=false
+CREATE TABLE users (id SERIAL PRIMARY KEY, name TEXT)
+INSERT INTO users (name) VALUES ('Alice')
+INSERT INTO users (name) VALUES ('Bob')
+```
+
+To exit `psql`, type `\q` and press enter.
+
+## The `/users` endpoint
+
+To talk to Postgres we'll need a client library. We're going to use `pg` for
+this tutorial. You can install it with:
+
+```shell showLineNumbers=false
+bun add pg @types/pg
+```
+
+To start with, we're going to hardcode our database connection details. Later
+on we'll learn how to use Effect to manage our app's configuration.
+
+Here's the code for our app in full, with a new `/users` endpoint added in
+Express:
+
+```ts twoslash ins={6,8,11-14,25} title=index.ts showLineNumbers=false
+import express from "express"
+
+import { NodeHttpServer, NodeRuntime } from "@effect/platform-node"
+import { HttpRouter, HttpServerResponse } from "@effect/platform"
+import { Effect } from "effect"
+import { Client } from "pg"
+
+const db = new Client({ user: "postgres" })
+const app = express()
+
+app.get("/users", async (req, res) => {
+  const { rows } = await db.query("SELECT * FROM users")
+  res.send({ users: rows })
+})
+
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/health", health)
+)
+
+const main = Effect.gen(function* () {
+  yield* Effect.promise(() => db.connect())
```

Comment:
```suggestion
  // Make sure we connect to the database before starting the server
  yield* Effect.promise(() => db.connect())
```

Might be worth adding a comment just for this  snippet since user's might be a bit confused why this is required? Idk